h2/src/main/org/h2/command/dml/Query.java

```diff
@@ -307,11 +307,7 @@ private boolean sameResultAsLast(Session s, Value[] params,
                 !isEverything(ExpressionVisitor.INDEPENDENT_VISITOR)) {
             return false;
         }
-        if (db.getModificationDataId() > lastEval &&
-                getMaxDataModificationId() > lastEval) {
-            return false;
-        }
-        return true;
+        return db.getModificationDataId() <= lastEval || getMaxDataModificationId() <= lastEval;
```

Comment:
What you're trying to optimize here and in many similar places?

Comparison is not a boolean, &&, ¦¦, etc are not booleans too in bytecode. Compiler will add the same code anyway. There is no win. You're only making source code less consistent. If some additional conditions will be required their insertion may be more complicated too.

Reply:
Compiler may even generate worse code in some cases with `return`, but it depends on used compiler.

Reply:
It's not really an optimization, as readability improvement. If you find this shorter code harder to read, I'll just revert all those changes.

Reply:
Such changes are OK when there is only one if-else operator that can be avoided at all.

But when there are additional `else if` blocks these changes only complicate reading and writing of the sources.

Reply:
I see your point. Would you rather I reverted those changes or get rid of the if/else in those situations and add appropriate formatting?

Reply:
This pull request will not be merged anyway. Send a new smaller requests.

In the new request you can change simple if-else (without else if) constructions.

But it's better not to touch large branches. Monster condition with a lot of || doesn't look good.

Reply:
I've reverted those changes.